src/shadowbox/server/server_access_key.ts

```diff
@@ -166,12 +175,30 @@ export class ServerAccessKeyRepository implements AccessKeyRepository {
     this.portForNewAccessKeys = port;
   }
 
-  async createNewAccessKey(encryptionMethod?: string): Promise<AccessKey> {
-    const id = this.keyConfig.data().nextId.toString();
+  private generateId(): string {
+    let id: AccessKeyId = this.keyConfig.data().nextId.toString();
     this.keyConfig.data().nextId += 1;
+    // Users can supply their own access key IDs. This means we always need to
+    // verify that any auto-generated key ID hasn't already been used.
+    while (this.isExistingAccessKeyId(id)) {
+      id = this.keyConfig.data().nextId.toString();
+      this.keyConfig.data().nextId += 1;
+    }
+    return id;
```

Comment:
Minor: I think the code is more readable if we remove the duplication, as it's clear the step action is the same as the initialization.
```suggestion
    // Users can supply their own access key IDs. This means we always need to
    // verify that any auto-generated key ID hasn't already been used.
    let id: AccessKeyId;
    while (id === undefined || this.isExistingAccessKeyId(id)) {
      id = this.keyConfig.data().nextId.toString();
      this.keyConfig.data().nextId += 1
    }
    return id
```